source/classes.tex

```diff
@@ -68,11 +68,12 @@
 An unnamed class thus can't
 be \tcode{final}.
 \end{note}
+Otherwise, the \grammarterm{class-name} is an \grammarterm{identifier};
+it is not looked up, and the \grammarterm{class-specifier} introduces it.
 
 \pnum
-A \grammarterm{class-name} is inserted into the scope in which it is
-declared immediately after the \grammarterm{class-name} is seen. The
-\grammarterm{class-name} is also inserted into the scope of the class
+The
+\grammarterm{class-name} is also bound in the scope of the class (template)
```

Comment:
"class (template)" wants to say "class or class template". We usually don't differentiate those two unless we must, because an instantiated class template is a class (and all class rules apply).
However, name lookup partially happens at template definition time, so we need to talk about the template stage of things.

Reply:
That seems like a good enough explanation. Please elaborate in #4403 if you think it's still not clear (I'm not creating an entry for it yet).